frontend/src/Components/NotificationsPopup.tsx

```diff
@@ -76,50 +79,80 @@ export default function NotificationsPopup(props: NotificationsPopupProps) {
-                        post={{ id: notify.source.post.id, site: notify.source.post.site }}
-                        commentId={notify.source.comment?.id}
-                        onClick={e => handleNotificationClick(e, notify)}
-                        onlyNew={true}
-                    >
-                        <div className={styles.type}>{notify.type === 'answer' ? <CommentIcon /> : <MentionIcon />}</div>
-                        <div className={styles.content}>
-                            <div className={styles.date}>{notify.source.byUser.username} {notify.type === 'answer' ? 'ответил вам' : 'упомянул вас'} <DateComponent date={notify.date} /></div>
-                            <div className={styles.text}>{notify.source.comment?.content}</div>
-                        </div>
-                    </PostLink>;
-                })}
-            </div>
-            <div className={styles.buttons}>
-                <button className={styles.buttonClear} onClick={handleClearAll}>Очистить</button>
-                <button className={styles.buttonAll} disabled={true}>Все чпяки</button>
+        <>
+            <div className={styles.overlay} onClick={props.onClose}/>
+            <div className={styles.container}>
+                <div className={styles.notifications}>
+                    {error}
+                    {notifications && notifications.map(notify => {
+                        return <PostLink
+                            key={notify.id}
+                            className={classNames(styles.notification, {[styles.read]: notify.read})}
+                            post={{ id: notify.source.post.id, site: notify.source.post.site }}
+                            commentId={notify.source.comment?.id}
+                            onClick={handleNotificationClick(notify)}
+                            onlyNew={true}
+                        >
+                            <div className={styles.type}>{notify.type === 'answer' ? <CommentIcon /> : <MentionIcon />}</div>
+                            <div className={styles.content}>
+                                <div className={styles.date}>{notify.source.byUser.username} {notify.type === 'answer' ? 'ответил вам' : 'упомянул вас'} <DateComponent date={notify.date} /></div>
+                                <div className={styles.text}>{notify.source.comment?.content}</div>
+                            </div>
+                            <div className={styles.remove} onClick={handleNotificationHide(notify)}><CloseIcon/></div>
+                        </PostLink>;
+                    })}
+                </div>
+                <div className={styles.buttons}>
+                    <button className={styles.buttonClear} onClick={handleClearAll}>Очистить</button>
+                    <button className={styles.buttonRead} onClick={handleReadAll}>Прочитать все</button>
+                    {/*<button className={styles.buttonAll} disabled={true}>Все чпяки</button>*/}
```

Comment:
remove

Reply:
Code and styles are still useful. Will mark as TODO.